builder/vsphere/driver/vm.go

```diff
@@ -970,11 +970,23 @@ func (vm *VirtualMachine) AddCdrom(controllerType string, isoPath string) error
 		return err
 	}
 
-	if isoPath != "" {
-		devices.InsertIso(cdrom, isoPath)
+	if datastoreIsoPath != "" {
+		isoPath := datastoreIsoPath
+		parts := strings.Split(datastoreIsoPath, "]")
+		if len(parts) > 1 {
+			// removes datastore name from path
+			isoPath = parts[1]
+		}
```

Comment:
I think it would be a good idea to extract that split feature here so that we can add tests on it. Something like:

```go
package vsphere

type IsoPath struct {
    Datastore string
    Path      string
}

func NewISOPath(path string) IsoPath {
  // ........
}